src/details/ArborX_DetailsTreeConstruction.hpp

```diff
@@ -364,25 +360,25 @@ class GenerateHierarchyFunctor
 
     if (split == first)
     {
-      _internal_nodes(i).children.first = &_leaf_nodes(split);
+      _internal_nodes(i).children.first = split + _shift;
```

Comment:
Please rename `_shift` to `_leaf_nodes_shift`.

Reply:
```diff
diff --git a/src/details/ArborX_DetailsTreeConstruction.hpp b/src/details/ArborX_DetailsTreeConstruction.hpp
index acc2c64..981b093 100644
--- a/src/details/ArborX_DetailsTreeConstruction.hpp
+++ b/src/details/ArborX_DetailsTreeConstruction.hpp
@@ -332,7 +332,7 @@ public:
       , _leaf_nodes(leaf_nodes)
       , _internal_nodes(internal_nodes)
       , _parents(parents)
-      , _shift(internal_nodes.extent(0))
+      , _leaf_nodes_shift(internal_nodes.extent(0))
   {
   }

@@ -360,8 +360,8 @@ public:

     if (split == first)
     {
-      _internal_nodes(i).children.first = split + _shift;
-      _parents(split + _shift) = i;
+      _internal_nodes(i).children.first = split + _leaf_nodes_shift;
+      _parents(split + _leaf_nodes_shift) = i;
     }
     else
     {
@@ -373,8 +373,8 @@ public:

     if (split + 1 == last)
     {
-      _internal_nodes(i).children.second = split + 1 + _shift;
-      _parents(split + 1 + _shift) = i;
+      _internal_nodes(i).children.second = split + 1 + _leaf_nodes_shift;
+      _parents(split + 1 + _leaf_nodes_shift) = i;
     }
     else
     {
@@ -388,7 +388,7 @@ private:
   Kokkos::View<Node *, DeviceType> _leaf_nodes;
   Kokkos::View<Node *, DeviceType> _internal_nodes;
   Kokkos::View<int *, DeviceType> _parents;
-  int _shift;
+  int _leaf_nodes_shift;
 };
```
I don't think that's a good change, it does not improve readability.  How about I add a comment to that effect next to the declaration of the data member?

Reply:
Disagree, I think it is easier to read. I would also do `_leaf_nodes_shift + <index>` to follow `offset + index` rule, but that's personal.